### DISCUSSION TITLE
Instances not being removed from the owner_manager, causing large amounts of memory usage

### DISCUSSION THREAD

**Discussion by smnrgrs** [OpenDDS/OpenDDS#4334] [Q&A]
We’ve been trying to trace resource leaks. In a test where we publish new instances at a high rate, we see memory increasing for publishers and subscribers, without dropping (the topic is Exclusive Ownership, Reliable, Dispose on Unregister). We take care to unregister each instance after publishing it as we will not publish any further updates to it.

We’ve found that when a new instance is added to the data reader we see it being added to the ownership_manager’s instance map. But when release_instance is called, this instance is not removed from ownership_manager’s map.

We’ve locally patched OpenDDS to remove the instance from owner_manager’s instance map in the release_instance_i call of DataReaderImpl_T.h. This seems to greatly improve the memory usage in our participants. 

Why don’t the instances get removed from the owner_manager?

Thanks

**Comment by jrw972** [comment]
This is a dusty corner of OpenDDS that I'm not familiar with.  However, I think it is telling that there are no statements that erase from  `type_instance_map_`, hence the memory leak you reported.  I think this is just a bug and oversight.  If you've already got a patch, please submit a PR.

**Reply by smnrgrs** [reply]
Thanks, we've opened PR #4343
